test/regressions/TestViewer.js

```diff
@@ -37,7 +37,10 @@ function MockTime(props) {
   React.useEffect(() => {
     // Use a "real timestamp" so that we see a useful date instead of "00:00"
     // eslint-disable-next-line react-hooks/rules-of-hooks -- not a React hook
-    clock = useFakeTimers(new Date('Mon Aug 18 14:11:54 2014 -0500'));
+    clock = useFakeTimers({
+      now: new Date('Mon Aug 18 14:11:54 2014 -0500').getTime(),
+      shouldAdvanceTime: true,
```

Comment:
Why this change?

Reply:
Otherwise, in the e2e test you fake the timers so you never apply the `setTimeout` of `debouncedResize` and the dimensions ref is never set.

Reply:
To give more context:
In `useGridDimensions` (and before in `useGridContainerResize`) we have the following logic:

```tsx
const isTestEnvironment = process.env.NODE_ENV === 'test';

[...]

if (isTestEnvironment) {
  // We don't need to debounce the resize for tests.
  resize();
  return;
}

debounceResize();
```

Which solves allow to test post-resize result synchronously in all tests except the `e2e` / `regressions`
In `e2e` tests we have no test impacted to my knowledge.
In `regressions` it causes issues because we never set the dimensions ref and therefore never triggers behavior like the `autoPageSize`.

I spent a few hours trying to fix this. I can't get Webpack to use `NODE_ENV === "test"` because it wants to have its mode equal to the `NODE_ENV` and it only has `development` and `production` modes.
So I just removed the mock of the timer, I am now only mocking the initial time to have reliable snapshots but not how the time evolve.

Reply:
>  I spent a few hours trying to fix this. I can't get Webpack to use NODE_ENV === "test" because it wants to have its mode equal to the NODE_ENV and it only has development and production modes.

It may be an useless question, but did you try `webpack.DefinePlugin`?

Reply:
Yes
If you put `NODE_ENV: "test"` it complains that the `NODE_ENV` should always be equal to the `mode`